osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/SecurityKeyProvider.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2021 Alliander N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+
+package org.opensmartgridplatform.adapter.protocol.dlms.domain.factories;
+
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.SecurityKeyType;
+
+/**
+ * Functional interface used in connectors to obtain keys.
+ *
+ * @see SecureDlmsConnector
+ */
+@FunctionalInterface
+public interface SecurityKeyProvider {
+  byte[] getKey(String deviceIdentification, SecurityKeyType type);
```

Comment:
Replacing this by `Map<SecurityKeyType, byte[]> getKeys(String deviceIdentification, List<SecurityKeyType> keyTypes)` looks like it could save a SOAP call to the secret management endpoint in connectors like the `Hls5Connector`.
The code in connectors with only a single secret type (like the `Lls1Connector`) would need to call with a singleton list or so, making it a little more complex. The benefits of not making additional SOAP calls in the other case might weigh in more (at least that is what I would expect).

Reply:
Replaced getKey with getKeys